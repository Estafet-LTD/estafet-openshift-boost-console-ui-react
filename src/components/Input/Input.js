import styled from 'styled-components';
import { ifProp } from 'styled-tools';

export default styled.input.attrs({ type: 'text' })`
  width: 100%;
  font-size: 0.9rem;
  outline: none;
  background: white;
  height: 2rem;
  line-height: 2rem;
  padding: 0 0.5rem;
  border: 1px solid #ced4da;
  border-radius: 3px;

  &::placeholder {
    color: #c8c8c8;
    font-size: 0.9rem;
  }

  &:read-only {
    color: #495057;
    background: #e9ecef;
    border-color: #ced4da;
  }

  ${ifProp('hasError', 'border-color: #dc3545;')};
`;
