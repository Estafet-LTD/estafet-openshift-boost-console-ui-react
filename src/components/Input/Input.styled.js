import styled from 'styled-components';

export const Wrapper = styled.input`
  font-size: 0.9rem;
  outline: none;
  background: white;
  height: 2rem;
  line-height: 2rem;
  padding: 0 0.5rem;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;

  &::placeholder {
    color: #c8c8c8;
    font-size: 0.9rem;
  }
`;
