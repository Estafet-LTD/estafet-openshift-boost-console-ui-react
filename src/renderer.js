// TODO: Clean up and split up all the different logic handling styled-components, variables set on window object, etc
import { ChunkExtractor, ChunkExtractorManager } from '@loadable/server';
import path from 'path';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';

import App from './App';
import configureStore from './store';

const store = configureStore();
const sheet = new ServerStyleSheet();

const statsFile = path.resolve('build/loadable-stats.json');
const extractor = new ChunkExtractor({
  statsFile,
  entrypoints: ['client'],
});

export default (req, res) => {
  const context = {};
  const html = renderToString(
    <ChunkExtractorManager extractor={extractor}>
      <StaticRouter context={context} location={req.url}>
        <Provider store={store}>
          <StyleSheetManager sheet={sheet.instance}>
            <App />
          </StyleSheetManager>
        </Provider>
      </StaticRouter>
    </ChunkExtractorManager>
  );

  const styleTags = sheet.getStyleTags();
  const scriptTags = extractor.getScriptTags();

  if (context.url) {
    res.redirect(context.url);
  } else {
    res.status(200).send(
      `<!doctype html>
      <html lang="">
        <head>
          <meta charset="utf-8" />
          <title>${process.env.PRODUCT_DESCRIPTION}</title>
          <meta name="description" content="" />
          <meta name="theme-color" content="#2BA2B8" />
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <link href="https://fonts.googleapis.com/css?family=Open+Sans&display=swap" rel="stylesheet">
          ${styleTags}
        </head>
        <body>
          <div id="root">${html}</div>
          ${scriptTags}
          <script>
            window.__PRELOADED_STATE__ = ${JSON.stringify(store.getState())};
            window.boost = ${JSON.stringify({
              BUILD_API_SERVICE_URI: process.env.BUILD_API_SERVICE_URI,
              TEST_API_SERVICE_URI: process.env.TEST_API_SERVICE_URI,
              PROD_API_SERVICE_URI: process.env.PROD_API_SERVICE_URI,
              PRODUCT_DESCRIPTION: process.env.PRODUCT_DESCRIPTION,
            })};
          </script>
        </body>
      </html>`
    );
  }
};
