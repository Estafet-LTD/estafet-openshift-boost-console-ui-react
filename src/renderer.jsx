import { ChunkExtractor, ChunkExtractorManager } from '@loadable/server';
import path from 'path';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';

import App from './App';
import configureStore from './store/configureStore';

const store = configureStore();
const sheet = new ServerStyleSheet();

// Mocking the loadable stats file for the tests
const isTesting = process.env.NODE_ENV === 'test';
const statsFileRoot = isTesting ? 'src/__mocks__' : 'build';
const statsFile = path.resolve(`${statsFileRoot}/loadable-stats.json`);

const extractor = new ChunkExtractor({
  statsFile,
  entrypoints: ['client'],
});

export default (req, res) => {
  const context = {};
  const html = renderToString(
    <ChunkExtractorManager extractor={extractor}>
      <StaticRouter context={context} location={req.url}>
        <Provider store={store}>
          <StyleSheetManager sheet={sheet.instance}>
            <App />
          </StyleSheetManager>
        </Provider>
      </StaticRouter>
    </ChunkExtractorManager>
  );

  const styledComponentsTags = sheet.getStyleTags();
  const styleTags = extractor.getStyleTags();
  const scriptTags = extractor.getScriptTags();

  if (context.url) {
    res.redirect(context.url);
  } else {
    res.status(200).send(
      `<!doctype html>
      <html lang="en">
        <head>
          <meta charset="utf-8" />
          <title>${process.env.PRODUCT_DESCRIPTION}</title>
          <meta name="description" content="" />
          <meta name="theme-color" content="#2BA2B8" />
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" rel="stylesheet">
          ${styleTags}
          ${styledComponentsTags}
        </head>
        <body>
          <div id="root">${html}</div>
          ${scriptTags}
          <script>
            window.__PRELOADED_STATE__ = ${JSON.stringify(store.getState())};
            window.__GLOBALS__= {
              PRODUCT: ${JSON.stringify(process.env.PRODUCT)},
              PRODUCT_DESCRIPTION: ${JSON.stringify(process.env.PRODUCT_DESCRIPTION)},
              TASK_MANAGEMENT_TITLE: ${JSON.stringify(process.env.TASK_MANAGEMENT_TITLE)},
              GATEWAY_API_SERVICE_URI: ${JSON.stringify(process.env.GATEWAY_API_SERVICE_URI)},
            };
          </script>
        </body>
      </html>`
    );
  }
};
