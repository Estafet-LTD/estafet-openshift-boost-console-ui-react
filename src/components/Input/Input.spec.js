import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';

describe('Components: Input', () => {
  it('it should match the snapshot [default]', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should match the snapshot [readonly]', () => {
    const tree = renderer.create(<Input readOnly />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should match the snapshot [hasError]', () => {
    const tree = renderer.create(<Input hasError />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
