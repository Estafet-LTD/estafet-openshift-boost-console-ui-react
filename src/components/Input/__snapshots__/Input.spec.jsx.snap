// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Input it should match the snapshot [default] 1`] = `
.c0 {
  width: 100%;
  font-size: 0.9rem;
  outline: none;
  background: white;
  height: 2rem;
  line-height: 2rem;
  padding: 0 0.5rem;
  border: 1px solid #ced4da;
  border-radius: 3px;
}

.c0::-webkit-input-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0::-moz-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0:-ms-input-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0::placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0:-moz-read-only {
  color: #495057;
  background: #e9ecef;
  border-color: #ced4da;
}

.c0:read-only {
  color: #495057;
  background: #e9ecef;
  border-color: #ced4da;
}

<input
  className="c0"
  type="text"
/>
`;

exports[`Components: Input it should match the snapshot [hasError] 1`] = `
.c0 {
  width: 100%;
  font-size: 0.9rem;
  outline: none;
  background: white;
  height: 2rem;
  line-height: 2rem;
  padding: 0 0.5rem;
  border: 1px solid #ced4da;
  border-radius: 3px;
  border-color: #dc3545;
}

.c0::-webkit-input-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0::-moz-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0:-ms-input-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0::placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0:-moz-read-only {
  color: #495057;
  background: #e9ecef;
  border-color: #ced4da;
}

.c0:read-only {
  color: #495057;
  background: #e9ecef;
  border-color: #ced4da;
}

<input
  className="c0"
  type="text"
/>
`;

exports[`Components: Input it should match the snapshot [readonly] 1`] = `
.c0 {
  width: 100%;
  font-size: 0.9rem;
  outline: none;
  background: white;
  height: 2rem;
  line-height: 2rem;
  padding: 0 0.5rem;
  border: 1px solid #ced4da;
  border-radius: 3px;
}

.c0::-webkit-input-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0::-moz-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0:-ms-input-placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0::placeholder {
  color: #c8c8c8;
  font-size: 0.9rem;
}

.c0:-moz-read-only {
  color: #495057;
  background: #e9ecef;
  border-color: #ced4da;
}

.c0:read-only {
  color: #495057;
  background: #e9ecef;
  border-color: #ced4da;
}

<input
  className="c0"
  readOnly={true}
  type="text"
/>
`;
